### PR TITLE
Optionally evaluate Groovy script if exists

### PR DESCRIPTION
**Pull request by marcphilipp**
In addition to the custom user data added via Java code in
`CustomUserData`, the extension now checks for a Groovy script in
`.mvn/gradle-enterprise-custom-user-data.groovy` and evaluates it if it
exists.

### CODE REVIEW

File: common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            throw new MavenExecutionException("Failed to evaluate custom user data Groovy script: " + scriptFile, e);
+        }
+    }
+
+    private static Binding prepareBinding(MavenSession session, BuildScanApi buildScan, Logger logger) {
+        Binding binding = new Binding();
+        binding.setVariable("project", session.getTopLevelProject());
+        binding.setVariable("session", session);
+        binding.setVariable("buildScan", buildScan);
+        binding.setVariable("log", new DefaultLog(logger));
```
Comment (marcphilipp) [inline common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java:45]: This is less than [what the GMavenPlus2 plugin provides](https://github.com/groovy/GMavenPlus/blob/a21bca3b086110d770ee639737edb37b89f34eb6/src/main/java/org/codehaus/gmavenplus/mojo/AbstractToolsMojo.java#L107-L144) but should be sufficient IMHO.

File: common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            logger.debug("Evaluating custom user data Groovy script: " + scriptFile);
+            evaluateGroovyScript(session, buildScan, logger, scriptFile);
+        } else {
+            logger.debug("Skipping evaluation of custom user data Groovy script because it does not exist: " + scriptFile);
+        }
+    }
+
+    private static File getScriptFile(MavenSession session) {
+        File rootDir = session.getRequest().getMultiModuleProjectDirectory();
+        return new File(rootDir, ".mvn/gradle-enterprise-custom-user-data.groovy");
```
Comment (snoopcheri) [inline common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java:28]: ❓ I guess it's enough when the script filename is fixed. But where/how will we document this to the user?

Comment (marcphilipp) [reply common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java:28]: We should probably add it [to the docs](https://docs.gradle.com/enterprise/maven-extension/#custom_extension) and the trial navigator.

Comment (facewindu) [reply common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java:28]: +1 I don't think we need to offer customization of this file

File: common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java (hunk 1 above)
Comment (facewindu) [reply common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java:45]: +1 we can always add more later

Comment (facewindu) [reply common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java:45]: but we should document what is available to the script, of course

**Comment by marcphilipp** [comment]
> 👍 just requested documentation updates to happen

@facewindu You mean what I wrote [here](https://github.com/gradle/gradle-enterprise-build-config-samples/pull/21#discussion_r440652394)?

File: common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java (hunk 1 above)
Comment (facewindu) [reply common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java:45]: already covered

**Pull request merged**